### ISSUE TITLE
12.0.3 RC2: merged-login.js temporarily (>multiple hours) empty after upgrade, leaving login button disabled

### ISSUE THREAD

**Issue by Fiech** [nextcloud/server#6540]
### Steps to reproduce
1. Update from 12.0.2 to 12.0.3 RC2
2. Navigate to login page and try to use login/use login button

### Expected behaviour
Login works as expected (either get into nextcloud or get to 2FA-Screen)

### Actual behaviour
Button/Enter-key not responsive. After multiple hours (at least 4-5h later, I rechecked not until 2 days later when I came around to writing this bug report) the login works as expected.

### Background information

I tracked down the problem to the fact that the JavaScript file /js/core/merged-login.js stayed empty for at least 4-5h or so. Everything else worked and I also could log in if I manuall remove the disabled-attribute in the dom-inspector. My hotfix was to remove the disabled-attribute in the template file.

Thing is: merged-share-backend.json (or /js/core/merged-share-backend.js) and merged-template-prepend.json (or /js/core/merged-template-prepend.js) worked just fine.

I compared the checksum of the involved files in merged-login.js with another, older NC installation, there is no difference.

I previously opened a [thread](https://help.nextcloud.com/t/12-0-3-rc2-merged-login-js-empty-leaving-login-button-disabled/21039) in the support forum.

I guess it's some kind of caching issue... Idk, why it only affected this one file...

### General server configuration

**Operating system:** Linux hermes 3.16.0-4-amd64 #1 SMP Debian 3.16.43-2+deb8u3 (2017-08-15) x86_64

**Web server:** nginx/1.12.1 (fpm-fcgi)

**Database:** pgsql PostgreSQL 9.4.13 on x86_64-unknown-linux-gnu, compiled by gcc (Debian 4.9.2-10) 4.9.2, 64-bit

**PHP version:** 7.0.23-1~dotdeb+8.1
<details>
        <summary>PHP-modules loaded</summary>

```
 - Core
 - date
 - libxml
 - openssl
 - pcre
 - zlib
 - filter
 - hash
 - Reflection
 - SPL
 - session
 - standard
 - cgi-fcgi
 - igbinary
 - PDO
 - xml
 - calendar
 - ctype
 - curl
 - dom
 - mbstring
 - fileinfo
 - ftp
 - gd
 - gettext
 - iconv
 - imagick
 - imap
 - json
 - exif
 - mcrypt
 - pdo_pgsql
 - pgsql
 - Phar
 - posix
 - readline
 - redis
 - shmop
 - SimpleXML
 - sockets
 - sysvmsg
 - sysvsem
 - sysvshm
 - tokenizer
 - wddx
 - xmlreader
 - xmlwriter
 - xsl
 - zip
 - Zend OPcache
```
</details>

### Nextcloud configuration

**Nextcloud version:** 12.0.3 RC2 - 12.0.3.1

**Updated from an older Nextcloud/ownCloud or fresh install:** YOUR ANSWER HERE

**Where did you install Nextcloud from:**  YOUR ANSWER HERE

**Are you using external storage, if yes which one:** Array
(
    [0] => \OC\Files\Storage\Local
    [1] => \OCA\Files_External\Lib\Storage\FTP
    [2] => \OC\Files\Storage\DAV
    [3] => \OCA\Files_External\Lib\Storage\OwnCloud
    [4] => \OCA\Files_External\Lib\Storage\SFTP
    [5] => \OCA\Files_External\Lib\Storage\AmazonS3
    [6] => \OCA\Files_External\Lib\Storage\Dropbox
    [7] => \OCA\Files_External\Lib\Storage\Google
    [8] => \OCA\Files_External\Lib\Storage\Swift
    [9] => \OCA\Files_External\Lib\Storage\SFTP
    [10] => \OCA\Files_External\Lib\Storage\SMB
    [11] => \OCA\Files_External\Lib\Storage\SMB
)

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** 

<details>
        <summary>Signing status</summary>

```
{
    "calendar": {
        "FILE_MISSING": {
            ".gitignore": {
                "expected": "2ebfa9d965d970eb2f3356f80cbff785c90c9a0d7634b406e49cd4e311826eb4abb482d8be75557cff491837881af2795fc4bd3a8e8d42b24c56d3b53b9b5dee",
                "current": ""
            }
        }
    }
}
```

This is discussed here https://github.com/nextcloud/calendar/issues/600

</details>

<details>
        <summary>Enabled apps</summary>

```
 - activity: 2.5.2
 - admin_audit: 1.2.0
 - admin_notifications: 1.0.0
 - bookmarks: 0.10.1
 - comments: 1.2.0
 - contacts: 1.5.3
 - dav: 1.3.0
 - encryption: 1.6.0
 - federatedfilesharing: 1.2.0
 - federation: 1.2.0
 - files: 1.7.2
 - files_external: 1.3.0
 - files_markdown: 2.0.1
 - files_pdfviewer: 1.1.1
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - files_videoplayer: 1.1.0
 - firstrunwizard: 2.1
 - gallery: 17.0.0
 - issuetemplate: 0.2.2
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - nextcloud_announcements: 1.1
 - notes: 2.3.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - password_policy: 1.2.2
 - provisioning_api: 1.2.0
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - spreed: 2.0.1
 - survey_client: 1.0.0
 - systemtags: 1.2.0
 - tasks: 0.9.5
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - twofactor_totp: 1.3.1
 - twofactor_u2f: 1.3.3
 - updatenotification: 1.2.0
 - workflowengine: 1.2.0
```
</details>

<details>
        <summary>Disabled apps</summary>

```
 - audioplayer
 - calendar
 - user_external
 - user_ldap
```
</details>

<details>
        <summary>Content of config/config.php</summary>

```
{
    "instanceid": "oc676d29f5b0",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "***REMOVED SENSITIVE VALUE***"
    ],
    "datadirectory": "\/owncloud\/data",
    "dbtype": "pgsql",
    "version": "12.0.3.1",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "localhost",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "forcessl": true,
    "theme": "",
    "maintenance": false,
    "secret": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "mail_from_address": "owncloud",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpauth": 1,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpsecure": "tls",
    "singleuser": false,
    "forceSSLforSubdomains": true,
    "loglevel": 2,
    "filelocking.enabled": "false",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "localhost",
        "port": 6379,
        "timeout": 0,
        "dbindex": 0
    },
    "trashbin_retention_obligation": "auto",
    "updater.release.channel": "beta",
    "overwrite.cli.url": "https:\/\/***REMOVED SENSITIVE VALUE***"
}
```
</details>

### Client configuration

**Browser:** Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/60.0.3112.113 Chrome/60.0.3112.113 Safari/537.36

**Operating system:** Ubuntu 16.10
### Logs

<details>
        <summary>Web server error log</summary>

```
No seemingly related lines, will of course provide if told what to look for
```
</details>

<details>
        <summary>Nextcloud log (data/nextcloud.log)</summary>

```
No seemingly related lines, will of course provide if told what to look for
```
</details>

**Comment by MorrisJobke** [comment]
cc @skjnldsv @juliushaertl

**Comment by rullzer** [comment]
I did not look into detail but I had this issue as well when upgrading:

Some of the things I checked:

* The file was present in the appdata_ dir
* The file contained data
* For some reason the file was empty when fetching

I have not been able to reproduce on my test env unfortunatly

**Comment by skjnldsv** [comment]
This is strange since the retrieving of the file is pretty simple code - wise... :/
Do you get a http code 200? Could you try wget the file?

**Comment by Fiech** [comment]
Unfortunately, I cannot test this anymore, but back then I checked with the error console in chromium and firefox and there was no error fetching the file clientside. The file simply was empty. I also navigated to the file with both browsers (with and without 'v'-GET-Parameter) only getting displayed an empty file.

**Comment by rullzer** [comment]
@skjnldsv I did get a 200 then. But i also could not figure out what was exactly causing it.

**Comment by Fiech** [comment]
So, I upgraded to 12.0.3 just now without any problems...

**Comment by skjnldsv** [comment]
Very strange.. :/

**Comment by MorrisJobke** [comment]
We cannot reproduce this anymore :/ I will close this ticket. If this happens again, just state it here and we can look into it. I guess #7109 also helps to mitigate such stuff.

**Comment by pellaeon** [comment]
Just upgraded to latest master and having the same issue, there are errors in the log, seems to be related to this issue?

```
{"reqId":"Vu9e55IcDhD10aAtodwi","level":3,"time":"2017-12-20T06:03:36+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/js\/core\/merged-login.js?v=1a9b45be-1","message":"file_get_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-login.js.gzip): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#214","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_get_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-template-prepend.js.deps): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#214","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_put_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-template-prepend.js): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#218","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_put_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-template-prepend.js.deps): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#218","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_put_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-template-prepend.js.gzip): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#218","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_get_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-share-backend.js.deps): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#214","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_put_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-share-backend.js): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#218","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_put_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-share-backend.js.deps): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#218","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_put_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-share-backend.js.gzip): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#218","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_get_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-login.js.deps): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#214","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_put_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-login.js): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#218","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_put_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-login.js.deps): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#218","userAgent":"UAUA","version":"13.0.0.8"}
{"reqId":"wg5Fx4jXm6MmQSOCRkDq","level":3,"time":"2017-12-20T06:03:37+00:00","remoteAddr":"IPIP","user":"--","app":"PHP","method":"GET","url":"\/login","message":"file_put_contents(\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/merged-login.js.gzip): failed to open stream: No such file or directory at \/usr\/home\/www\/beta\/lib\/private\/Files\/Storage\/Local.php#218","userAgent":"UAUA","version":"13.0.0.8"}
```

**Comment by SteffenDE** [comment]
I am experiencing the same error:
```
{"reqId":"HCHmhw0ptiiaVC9NKXy6","level":3,"time":"2018-01-25T18:52:38+00:00","remoteAddr":"","user":"--","app":"PHP","method":"GET","url":"\/index.php\/js\/core\/merged-template-prepend.js?v=795158a7e9fd2348d10fa94579fd7d0b-2","message":"fread() expects parameter 1 to be resource, null given at \/var\/www\/virtual\/nextcloud\/lib\/private\/Files\/Storage\/Wrapper\/Encryption.php#850","userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/63.0.3239.132 Safari\/537.36","version":"12.0.5.3"}
{"reqId":"HCHmhw0ptiiaVC9NKXy6","level":3,"time":"2018-01-25T18:52:38+00:00","remoteAddr":"","user":"--","app":"PHP","method":"GET","url":"\/index.php\/js\/core\/merged-template-prepend.js?v=795158a7e9fd2348d10fa94579fd7d0b-2","message":"fclose() expects parameter 1 to be resource, null given at \/var\/www\/virtual\/nextcloud\/lib\/private\/Files\/Storage\/Wrapper\/Encryption.php#851","userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/63.0.3239.132 Safari\/537.36","version":"12.0.5.3"}
{"reqId":"lPKAjrDAmcFKoMPJqteE","level":3,"time":"2018-01-25T18:52:40+00:00","remoteAddr":"","user":"--","app":"PHP","method":"GET","url":"\/index.php\/js\/core\/merged-template-prepend.js","message":"fread() expects parameter 1 to be resource, null given at \/var\/www\/virtual\/nextcloud\/lib\/private\/Files\/Storage\/Wrapper\/Encryption.php#850","userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/63.0.3239.132 Safari\/537.36","version":"12.0.5.3"}
{"reqId":"lPKAjrDAmcFKoMPJqteE","level":3,"time":"2018-01-25T18:52:40+00:00","remoteAddr":"","user":"--","app":"PHP","method":"GET","url":"\/index.php\/js\/core\/merged-template-prepend.js","message":"fclose() expects parameter 1 to be resource, null given at \/var\/www\/virtual\/nextcloud\/lib\/private\/Files\/Storage\/Wrapper\/Encryption.php#851","userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/63.0.3239.132 Safari\/537.36","version":"12.0.5.3"}
{"reqId":"DcWgzjbYP88dXtp4SxrY","level":3,"time":"2018-01-25T18:52:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"GET","url":"\/index.php\/js\/core\/merged-template-prepend.js","message":"fread() expects parameter 1 to be resource, null given at \/var\/www\/virtual\/nextcloud\/lib\/private\/Files\/Storage\/Wrapper\/Encryption.php#850","userAgent":"HTTPie\/0.9.9","version":"12.0.5.3"}
{"reqId":"DcWgzjbYP88dXtp4SxrY","level":3,"time":"2018-01-25T18:52:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"GET","url":"\/index.php\/js\/core\/merged-template-prepend.js","message":"fclose() expects parameter 1 to be resource, null given at \/var\/www\/virtual\/nextcloud\/lib\/private\/Files\/Storage\/Wrapper\/Encryption.php#851","userAgent":"HTTPie\/0.9.9","version":"12.0.5.3"}
{"reqId":"TxJRWga84v4e7OWjT8Px","level":3,"time":"2018-01-25T18:54:37+00:00","remoteAddr":"","user":"--","app":"PHP","method":"GET","url":"\/index.php\/js\/core\/merged-template-prepend.js","message":"fread() expects parameter 1 to be resource, null given at \/var\/www\/virtual\/nextcloud\/lib\/private\/Files\/Storage\/Wrapper\/Encryption.php#850","userAgent":"HTTPie\/0.9.9","version":"12.0.5.3"}
{"reqId":"TxJRWga84v4e7OWjT8Px","level":3,"time":"2018-01-25T18:54:37+00:00","remoteAddr":"","user":"--","app":"PHP","method":"GET","url":"\/index.php\/js\/core\/merged-template-prepend.js","message":"fclose() expects parameter 1 to be resource, null given at \/var\/www\/virtual\/nextcloud\/lib\/private\/Files\/Storage\/Wrapper\/Encryption.php#851","userAgent":"HTTPie\/0.9.9","version":"12.0.5.3"}
```

**Comment by pellaeon** [comment]
Last time I have tried to debug, I found this error related to the SimpleFS layer that the appdata utilize.

From my logs, basically it was that appdata failed to use SimpleFS to create the directory `\/usr\/home\/www\/beta\/data\/appdata_ocijq7g01thu\/js\/core\/`, so the CSS and JS preprocessors failed to write the gzipped files to that dir.

However, if you change instanceid, a new appdata folder with new instanceid will be created without problem.

What I was not able to understand is that, from SimpleFS's documentations, it seems only to be used for managing **files within one single folder**, no nesting is allowed, but somehow appdata still uses SimpleFS to create nested folders (`js/core`), why doesn't appdata just use the standard storage driver instead?

**Comment by skjnldsv** [comment]
@SteffenDE are you using encryption?
Thanks for reporting this issue though, we'll take a look at it really soon!

**Comment by SteffenDE** [comment]
@skjnldsv Yes, encryption is enabled. This error occurs every time I try to update from 12.0.4.3 to 12.0.5.

**Comment by juliushaertl** [comment]
@rullzer Anything with the appdata that might be break when encryption is enabled?

**Comment by rullzer** [comment]
Should not. It isn't a user folder so it doesn't write them encrypted I think.

**Comment by rullzer** [comment]
@pellaeon it is an abstraction on top of the regular FS. Because initially the idea was to be able to move this to fast storage and keep the rest of data on HDD.

But it uses the normal Node API.

Also, each app gets a folder (so in this case `js`) is considered an app. Ther you can then create 1 layer of subfolders. And in those folders files.

**Comment by rullzer** [comment]
I just wrote a small app to help here:

Could you try:

1. cd apps
2.wget https://github.com/rullzer/appdatacleaner/releases/download/v1.0.0/appdatacleaner.tar.gz
3. tar -xzvf appdatacleaner.tar.gz
4. cd ..
5. ./occ app:enable appdatacleaner:
6. ./occ appdatacleaner:css
7. ./occ appdatacleaner:js
8. Go to your cloud in your browser and reload

**Comment by SteffenDE** [comment]
![bildschirmfoto 2018-02-03 um 13 18 17](https://user-images.githubusercontent.com/4116351/35767072-e2fb825e-08e4-11e8-93a8-7bee524f2d1b.png)

I think I forgot to mention that we are using OpenStack ObjectStorage as primary storage in Nextcloud. I think your script @rullzer broke some css files as I got many `{\"Exception\":\"OpenCloud\\\\ObjectStore\\\\Exception\\\\ObjectNotFoundException\"` errors after using it. I had to manually upload the files (by urn) to the ObjectStore to get it working again.

By now I think that using an ObjectStore with Nextcloud is a bad idea:
```
{"reqId":"89Mp9DXVjm3nWXBDaJcL","level":3,"time":"2018-02-03T14:07:28+00:00","remoteAddr":"","user":"Administrator","app":"objectstore","method":"GET","url":"\/index.php\/core\/preview.png?file=\/Photos\/Coast.jpg&c=5a75c21d71918&x=150&y=150","message":"Count not get object urn:oid:10957 for file appdata_ocwsoj076el8\/preview\/1479\/1100-734-max.png: {\"Exception\":\"OpenCloud\\\\ObjectStore\\\\Exception\\\\ObjectNotFoundException\",\"Message\":\"urn:oid:10957 could not be found. The API returned this HTTP response:\\n\\nHTTP\\\/1.1 404 Not Found
```
Sadly, I didn't find an option to migrate back to using normal storage.

Would it be possible to NOT save the appdata files in the ObjectStore?

**Comment by rullzer** [comment]
mmm I have don't get why that would happen. The app indeed removes all the generated css and js files. But that is what it is supposed to do. Your instance should regenerate them.

**Comment by juliushaertl** [comment]
I guess it's the same as in https://github.com/nextcloud/server/issues/8041

**Comment by SteffenDE** [comment]
@juliushaertl yep, that's it. Still doesn't explain why my css was screwed up after using the appdata cleaner.

**Comment by tflidd** [comment]
Some more app-data-issues for a few people upgrading to NC 13:
https://help.nextcloud.com/t/no-way-to-login-and-enter-in-my-nexcloud-13-update/26957

**Comment by juliushaertl** [comment]
@SteffenDE There was a bug in the swift implementation. See https://github.com/nextcloud/server/issues/8041#issuecomment-364034230 for details.

Fixed with #8233
